proto/config/config.proto

```diff
@@ -19,11 +19,14 @@ message Global {
   // If present, configures identity.
   IdentityContext identity_context = 4;
 
+  // Stores information about the last installation/upgrade.
+  //
+  // Useful for driving upgrades.
+  InstallContext install_context = 5;
```

Comment:
I could go either way on this, but the original idea behind the `linkerd-config` ConfigMap was that it would have multiple keys corresponding to multiple different types of config structs. At the moment, we have keys for "global" and "proxy", corresponding to the `Global` and `Proxy` structs in this file. Since the new config that you're adding is only relevant to install (and upgrade), I'd be inclined to put it in a separate `Install` struct, as follows:

```diff
diff --git a/proto/config/config.proto b/proto/config/config.proto
index 4e365751..a7e92a83 100644
--- a/proto/config/config.proto
+++ b/proto/config/config.proto
@@ -9,6 +9,7 @@ option go_package = "github.com/linkerd/linkerd2/controller/gen/config";
 message All {
   Global global = 1;
   Proxy proxy = 2;
+  Install install = 3;
 }
 
 message Global {
@@ -19,11 +20,6 @@ message Global {
   // If present, configures identity.
   IdentityContext identity_context = 4;
 
-  // Stores information about the last installation/upgrade.
-  //
-  // Useful for driving upgrades.
-  InstallContext install_context = 5;
-
   // If present, indicates that the Mutating Webhook Admission Controller should
   // be configured to automatically inject proxies.
   AutoInjectContext auto_inject_context = 6;
@@ -74,7 +70,7 @@ message IdentityContext {
   google.protobuf.Duration clock_skew_allowance = 4;
 }
 
-message InstallContext {
+message Install {
   // The unique ID fr this installation. Does not change on upgrade.
   string uuid = 1;
 
```

Granted, we don't have a lot of precedent around the organization of this file just yet. But if we're going to treat `Global` as a grab bag of configuration options, I'd be inclined to do away with the multiple config struct setup, and instead move the proxy config options into the global struct as well. But if we're going to try to separate out different types of a configuration, this would be a good opportunity to add a third, top-level `Install` struct.

At the end of the day, I don't have a strong preference, but I want to take whichever approach avoids an unmanageable mess of configuration options :)

Reply:
Yeah, this seems like a good idea. I'll incorporate this once i have the upgrade branch sorted out a bit further. Thanks!

Reply:
I guess the main counterpoint to this is that it will have to get added to `config.All`, so it will still be shipped around everywhere. But conceptually, I think it's probably still worth it...